experimental/InvariantReductiveGroups/InvariantRings_ReducetiveGroups.jl

```diff
@@ -0,0 +1,478 @@
+        z.poly_ring, _ = grade(PolynomialRing(z.field, "X" => 1:n)[1])
+        z.group = (sym, m)
+        M = matrix(R,1,m,gens(R)[1:m])
+        for i in 1:m-1 
+            M = vcat(M, matrix(R,1,m,gens(R)[(i)*m+1:(i+1)*m]))
+        end
+        det_ = det(M)
+        z.group_equations = det_ - 1
+        z.group_rep = rep_mat
+        z.generators = inv_generators(rep_mat, z.poly_ring, det_)
+        return z
+    end
+    
+    # degree
+    function InvariantRing(sym::Symbol, m::Int64, sym_deg::Int64)
+        #sym != SL && return nothing
+        z = new()
+        z.field = QQ
+        z.group_rep = rep_mat_(m, sym_deg)
+        n = ncols(z.group_rep)
+        z.poly_ring, __ = grade(PolynomialRing(z.field, "X" => 1:n)[1])
+        z.group = (sym, m)
+        group_poly_ring, Z = PolynomialRing(QQ, "Z"=>(1:m,1:m))
+        M = matrix(group_poly_ring,m,m,[Z[i,j] for i in 1:m, j in 1:m])
+        det_ = det(M)
+        z.group_equations = det_ - 1
+        z.generators = inv_generators(z.group_rep, z.poly_ring, det_)
+        return z
+    end
+    
+    #direct sum
+    function InvariantRing(sym::Symbol, m::Int64, v::Vector{Int64}, sum::Bool = true)
+        #sym != SL && return nothing
+        z = new()
+        z.field = QQ 
+        z.group_rep = rep_mat_(m, v, sum)  
+        n = ncols(z.group_rep)
+        z.poly_ring, __ = grade(PolynomialRing(z.field, "X" => 1:n)[1])
+        z.group = (sym, m)
+        group_poly_ring, Z = PolynomialRing(QQ, "Z"=>(1:m,1:m))
+        M = matrix(group_poly_ring,m,m,[Z[i,j] for i in 1:m, j in 1:m])
+        det_ = det(M)
+        z.group_equations = det_ - 1
+        z.generators = inv_generators(z.group_rep, z.poly_ring, det_)
+        return z
+    end
+    
+    function InvariantRing(sym::Symbol, m::Int64, sym_deg::Int64, prod::Int64)
+        #sym != SL && return nothing
+        z = new()
+        z.field = QQ
+        rep_mat_ = rep_mat_(m, sym_deg)
+        for i in 0:prod-1
+            rep_mat_ = kroenecker_product(rep_mat_,rep_mat_)
+        end
+        z.group_rep = rep_mat_
+        n = ncols(z.group_rep)
+        z.poly_ring, __ = grade(PolynomialRing(z.field, "X" => 1:n)[1])
+        z.group = (sym, m)
+        group_poly_ring, Z = PolynomialRing(QQ, "Z"=>(1:m,1:m))
+        M = matrix(group_poly_ring,m,m,[Z[i,j] for i in 1:m, j in 1:m])
+        det_ = det(M)
+        z.group_equations = det_ - 1
+        z.generators = inv_generators(z.group_rep, z.poly_ring, det_)
+        return z
+    end
+end
+
+function gens(R::InvariantRing)
+    return R.generators
+end
+
+
+function Base.show(io::IO, R::InvariantRing) #TODO compact printing
+    #if get(io, :supercompact, false)
+        print(io, "Invariant Ring of", "\n")
+        show(io, R.poly_ring)
+        print(io, " under group action of ", R.group[1], R.group[2], "\n", "\n")
+        print(io, "Generated by ", "\n")
+    for i in 1:length(R.generators)
+        show(io, R.generators[i])
+        print(io, "\n")
+    end
```

Comment:
Perhaps this would work
```suggestion
    join(io, R.generators, "\n")
```